vexim/config/functions.php

```diff
@@ -127,11 +127,28 @@ function crypt_password($clear, $salt = '')
                     }
                     $cryptedpass = crypt($clear, $salt);
                 break;
+		case 'sha512':
+                    if (empty($salt))
+                    {
+                        $salt = '$6$'.get_random_bytes(16).'$';
+                    }
+                    $cryptedpass = crypt($clear, $salt);
+		break;
+		case 'bcrypt':
+                    if (empty($salt))
+                    {
+                        $salt = '$2a$10$'.get_random_bytes(22).'$';
+                    }
+                    $cryptedpass = crypt($clear, $salt);
+		break;
```

Comment:
I'd like to consolidate this function as much as possible.
From my understanding, we don't really have to know the length of the salt each time, since the algorithm in use will simply ignore all unnecessary characters. Nor does the salt have to end with the dollar sign. Thus, I think we could set only the salt prefix in this `switch` block, and only do that when the salt itself is empty. After setting the prefix, we could run `get_random_bytes()` with `CRYPT_SALT_LENGTH` as an argument.
Only exceptions to this scheme would be the algorithms which don't use `crypt()`, namely, `sha` and `clear`.
Also, the default case statement below could have a simpler regex used in `preg_match()` in this case: something like `/^\$[:digit:][:alnum:]?\$/`. I don't think we should enforce only `$2*$` and `$6$` here.